### PR TITLE
Match disabled apps and update sort method

### PR DESCRIPTION
**Pull request by shadowsocksRb**
对 #2335 的后续补充

### CODE REVIEW

File: mobile/src/main/java/com/github/shadowsocks/AppManager.kt [hunk 1]
```diff
@@ -79,7 +82,7 @@ class AppManager : AppCompatActivity() {
             }
             // Labels and icons can change on configuration (locale, etc.) changes, therefore they are not cached.
             val cachedApps = cachedApps ?: pm.getInstalledPackages(
-                    PackageManager.GET_PERMISSIONS or PackageManager.MATCH_UNINSTALLED_PACKAGES)
+                    GET_PERMISSIONS or MATCH_UNINSTALLED_PACKAGES or MATCH_DISABLED_COMPONENTS)
```
Comment (Mygod) [inline mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: Why is matching components useful for getting packages?

File: mobile/src/main/java/com/github/shadowsocks/AppManager.kt [hunk 2]
```diff
@@ -135,7 +138,11 @@ class AppManager : AppCompatActivity() {
             apps = getCachedApps(packageManager).map { (packageName, packageInfo) ->
                 coroutineContext[Job]!!.ensureActive()
                 ProxiedApp(packageManager, packageInfo.applicationInfo, packageName)
-            }.sortedWith(compareBy({ !isProxiedApp(it) }, { it.name.toString() }))
+            }.sortedWith(compareBy(
+                    { !isProxiedApp(it) },
+                    { it.appInfo.flags and ApplicationInfo.FLAG_SYSTEM },
+                    { it.name.toString() }
+            ))
```
Comment (Mygod) [inline mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: I consider this change to be confusing to users. Please do not put two (or more) unrelated changes in a single pull request in the future.

File: mobile/src/main/java/com/github/shadowsocks/AppManager.kt (hunk 1 above)
Comment (shadowsocksRb) [reply mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: AOSP没有`MATCH_DISABLED_PACKAGES`，这个flag用在这里就是匹配已停用应用。
这个问题需要问Google才有答案

File: mobile/src/main/java/com/github/shadowsocks/AppManager.kt (hunk 2 above)
Comment (shadowsocksRb) [reply mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: 还有这种理由？！用户会疑惑为什么某某应用没有显示，括号里的数字是什么，为什么有些数字一样。但唯独不在乎排序乱了：能找到就行，快点更好。

File: mobile/src/main/java/com/github/shadowsocks/AppManager.kt (hunk 1 above)
Comment (Mygod) [reply mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: You should provide relevant documentations/blog posts/Android platform source code to back up your claim. You should not expect me to read through the source code of `PackageManagerService.java` to find out whether this flag is indeed useful, or it is just a coding mistake someone else made online and you just copied their code mindlessly.

File: mobile/src/main/java/com/github/shadowsocks/AppManager.kt (hunk 2 above)
Comment (Mygod) [reply mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: I am still unconvinced as there are also numerous other sorting options that are still as valid *and* as arbitrary as this. By Occam's razor I prefer the old way.

@madeye can weigh in on this.

Comment (madeye) [reply mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: We should avoid this change.

It's intuitive that we put selected apps on the top. However, if we keep adding other sorting options here, then users will get confused easily.

File: mobile/src/main/java/com/github/shadowsocks/AppManager.kt (hunk 1 above)
Comment (shadowsocksRb) [reply mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: https://android.googlesource.com/platform/frameworks/base/+/a9ef0e9bb4df194ea5a2575c8ef0fe8d1572b1f4/core/java/android/content/pm/PackageManager.java#176
同样讨厌去pm service里寻找具体逻辑。这个可以直接在AS跳转源码里看见，`ApplicationInfoFlags`注解里包括了`MATCH_DISABLED_COMPONENTS`，这应该很明显。

Comment (Mygod) [reply mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: `getInstalledPackages` returns a list of `PackageInfo` that can contain other `ComponentInfo`. The documentation of `MATCH_DISABLED_COMPONENTS` reads `flag: include disabled components in the returned info`. As far as I see, this flag serves no purpose for this.

Comment (shadowsocksRb) [reply mobile/src/main/java/com/github/shadowsocks/AppManager.kt]: …………
这些代码已经在rb里用上了，行为符合预期，一切良好。

**Comment by Mygod** [comment]
I stand with my original decisions in #2335. Just a reminder that *please* do not put two (or more) unrelated changes in a single pull request in the future. Having one change for every pull request would make the reviewing much easier and we can merge approved changes much faster. Thanks again.

**Pull request closed without merging**